viz_scripts/generic_timeseries.ipynb

```diff
@@ -154,26 +157,48 @@
     "        mode_counts.rename(columns={'size':'trip_count'}, inplace=True)\n",
     "\n",
-    "        # Sum daily distance traveled for each mode\n",
-    "        mode_distance = data.groupby(['user_id','date_time','Mode_confirm'], as_index=False)[['distance']].sum()\n",
-    "        mode_distance.rename(columns={'sum':'distance'}, inplace=True)\n",
-    "        mode_distance['distance'] = mode_distance['distance'].fillna(0)\n",
-    "\n",
-    "        # Sum daily emissions for each user\n",
-    "        emissions = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_kg_CO2', 'distance']].sum()\n",
-    "        emissions['Mode_confirm_kg_CO2'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
-    "        emissions['distance'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
-    "\n",
-    "        # Sum daily energy for each user\n",
-    "        energy = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_EI(kWH)', 'distance']].sum()\n",
-    "        energy['Mode_confirm_EI(kWH)'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
-    "        energy['distance'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
-    "\n",
-    "        # Add 7-day rolling avg smoothing to better see trends\n",
-    "        mode_counts['trip_count_smooth'] = mode_counts.groupby(['user_id','Mode_confirm'])['trip_count'].apply(lambda x: x.rolling(7,1).mean())\n",
-    "        mode_distance['distance_smooth'] = mode_distance.groupby(['user_id','Mode_confirm'])['distance'].apply(lambda x: x.rolling(7,1).mean())\n",
-    "        emissions['distance_smooth'] = emissions.groupby(['user_id'])['distance'].apply(lambda x: x.rolling(7,1).mean())\n",
-    "        energy['distance_smooth'] = energy.groupby(['user_id'])['distance'].apply(lambda x: x.rolling(7,1).mean())"
+    "        if (len(dynamic_labels)) > 0:\n",
+    "            # Sum daily distance traveled for each mode\n",
+    "            mode_distance = data.groupby(['user_id','date_time','Mode_confirm'], as_index=False)[['distance']].sum()\n",
+    "            mode_distance.rename(columns={'sum':'distance'}, inplace=True)\n",
+    "            mode_distance['distance'] = mode_distance['distance'].fillna(0)\n",
+    "\n",
+    "            # Sum daily emissions for each user\n",
+    "            emissions = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_kg_CO2', 'distance']].sum()\n",
+    "            emissions['Mode_confirm_kg_CO2'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
+    "            emissions['distance'] = emissions['Mode_confirm_kg_CO2'].fillna(0)\n",
+    "\n",
+    "            # Sum daily energy for each user\n",
+    "            energy = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_EI(kWH)', 'distance']].sum()\n",
+    "            energy['Mode_confirm_EI(kWH)'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "            energy['distance'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
```

Comment:
we don't have energy computations for the dynamic labels, right?
the custom labels don't include them.

Reply:
Yes, We're not showing the energy computations for the dynamic labels. It makes sense to remove the computation of energy related part from here.